android/capacitor/src/main/java/com/getcapacitor/CapConfig.java

```diff
@@ -9,21 +12,82 @@
 import org.json.JSONObject;
 
 /**
- * Management interface for accessing values in capacitor.config.json
+ * Represents the configuration options for Capacitor
  */
 public class CapConfig {
 
-    private JSONObject config = new JSONObject();
+    // Server Config
+    private boolean html5mode = true;
+    private String serverUrl;
+    private String hostname = "localhost";
+    private String androidScheme = CAPACITOR_HTTP_SCHEME;
+    private String[] allowNavigation;
+
+    // Android Config
+    private String overriddenUserAgentString;
+    private String appendedUserAgentString;
+    private String backgroundColor;
+    private boolean allowMixedContent = false;
+    private boolean captureInput = false;
+    private Boolean webContentsDebuggingEnabled;
+    private boolean hideLogs = false;
+
+    // Plugins
+    private JSONObject pluginConfigurations = new JSONObject();
 
-    public CapConfig(AssetManager assetManager, JSONObject config) {
-        if (config != null) {
-            this.config = config;
-        } else {
-            // Load our capacitor.config.json
-            this.loadConfig(assetManager);
+    // Config Object JSON (legacy)
+    private JSONObject configJSON = new JSONObject();
+
+    /**
+     * Constructs a Capacitor Configuration from file.
+     *
+     * @param context The current context
+     * @param defaultDebuggable Enable dev mode flag
+     */
+    public CapConfig(Context context, boolean defaultDebuggable) {
+        if (context == null) {
+            Logger.error("Capacitor Config could not be created. Context must not be null.");
+            return;
         }
+
+        AssetManager assetManager = context.getAssets();
+        webContentsDebuggingEnabled = defaultDebuggable;
+
+        // Load capacitor.config.json
+        loadConfig(assetManager);
+        deserializeConfig();
```

Comment:
Should all this be happening in the constructor? Maybe we can take this opportunity to practice some separation of concerns. The iOS side now has a few different classes that separate "loading from a JSON file" and "containing the configuration".

Reply:
Thanks for the feedback. I have removed the public constructor for CapConfig and replaced it with a method `fromFile` that can be used to create an instance of CapConfig from the JSON config file. 

I moved the JSON file loading to the FileUtils class since it involved reading a file and seemed appropriate there (and its also reused elsewhere in the core).